[16.0][MIG] - shopfloor

sbejaoui opened the pull request:
needs:

- #642 
- https://github.com/OCA/stock-logistics-workflow/pull/1330
- https://github.com/OCA/product-attribute/pull/1215
- https://github.com/OCA/stock-logistics-workflow/pull/1469

sebalix committed:
shopfloor, delivery: new option 'Process as pre-packaged' products

This option allows to scan product packaging for bulk products
and process the packaging qty.

mmequignon committed:
shopfloor, cluster & zone picking: put multiple moves on a same dest. package

When picking a move, allow to set a destination package that was already used
for the other lines.

mmequignon committed:
shopfloor, cluster & zone picking: unload dest. package at destination

With this option, the lines you process by putting on a package during the
picking process will be put as bulk products at the final destination location.

This is useful if your picking device is emptied at the destination location or
if you want to provide bulk products to the next operation.

Incompatible with: "Pick and pack at the same time"

mmequignon committed:
shopfloor, checkout: Allow to scan products

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

sebalix committed:
shopfloor, delivery: force qty to process to 1 when scanning a product

Commit:
[UPD] Update shopfloor.pot

sebalix committed:
shopfloor: fix user comparison in 'mark_move_lines_as_picked'

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.3.1

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.4.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.5.0

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (230 of 230 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

mmequignon committed:
shopfloor search: split product_from_scan

sebalix committed:
shopfloor, single pack transfer: remove useless code

No need to generate moves manually for the package level, it is done
automatically by calling `action_confirm` on its picking right after.

sebalix committed:
shopfloor, single pack transfer: set putaway dest. for package levels

It is already working for package levels containing only one
product/move, but it doesn't for packages having several products.

sebalix committed:
shopfloor, single pack transfer: fix cancel action for mixed packages

Fix 'Expected singleton' error when calling the cancel action.
As a package could contain multiple products, a package level could be
linked to multiple moves.

TDu committed:
Add sublocation scan on shopfloor delivery scenario

When scanning a product barcode or a lot the scenario will search
for all related move lines and pick the first one by default.

For a shopfloor menu configuration with locations containing sub locations
this is not ideal. Because a user working at a sub location only wants
move lines related to that sub location to be selected.

To help with this situation, a user can first scan his working location
which is kept as long as no other location is scanned.
The sub location set can also be reset by the click of a button.

Also when the search for a move line to work on returns multiple move lines
related to different locations and no location has been preliminary scanned.
The user will be prompted to scann a location first.

OCA-git-bot committed:
shopfloor 14.0.2.5.1

TDu committed:
shopfloor: change search location action

Change the search to first look on the barcode field and then on
location name.

TDu committed:
shopfloor: add checkout scenario uml graph

TDu committed:
shopfloor: zone picking scenario stay in zone

When scanning a location out of the previously selected zone, instead of
sending the user back to the start, keep him on the line selection of
the zone.

TDu committed:
shopfloor: zone picking add unloading on first screen

The user may end up in the `scan_location` screen with some products to
unload. It can happen when clicking on the back button but also when
there is no lines left to process in the current zone.

This change adds an Unload At Destination button on the `scan_location`
screen which will be display only when the user has some product to
unload.

OCA-git-bot committed:
shopfloor 14.0.2.6.0

OCA-git-bot committed:
shopfloor 14.0.2.7.0

sebalix committed:
shopfloor: imp zp scan source package message

TDu committed:
shopfloor: imp zp scan source location

Fix the error raised when there are multiple lots in location.
In case of error, reduce the list to the new location scanned
and displayed lines are filtered accordingly.

TDu committed:
shopfloor: imp zp scan source product

Ff multiple move lines are found on several locations, raise an error
and filter the list with the matched moves.

If multiple move lines are found for different lots, raise an error
and filter the list with the matched moves.

TDu committed:
shopfloor: imp zp scan source lot

If multiple move lines are found on several locations, display a message
and filter the list with the matched moves.
If multiple move lines are found for different products, raise an error "Several products found, please scan first a product"

TDu committed:
shopfloor: imp cluster picking  _scan_line_by_package

TDu committed:
shopfloor: fix cluster picking _scan_line_by_product

Check the content of the location not the move lines of the picking.

TDu committed:
shopfloor: fix cluster picking _scan_line_location

Fix the error raised if there are multiple packs, products or lot in the location.
Not based on move lines but based on actual stock at location.

TDu committed:
shopfloor: fix cluster picking _scan_line_lot

Take into account the content of the location not move on the picking
when checking for lot in multiple packages.

TDu committed:
shopfloor: fix test multiple lot,product,pack on location

The tests were like they were because the code was testing the existance
of a lot, pack or product in a location based on existing pick/moves.
But now the code tests the location directely for such information.

TDu committed:
shopfloor, delivery: change processed quantiy on product scan

Until now a line was always fully delivered.

With this change

For move lines without pacakge:

* When scanning a product only a quantity of one should be processed.
* When scanning a packaging process the quantity of the packaging.

For move lines with a package:

* When scanning a product only if the line package contains one unity of
  the product and that product only.

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.8.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.9.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

sebalix committed:
shopfloor, delivery: add a test related to 'Process as pre-packaged' option

sebalix committed:
shopfloor, delivery: process lines of one delivery operation

If the same bulk product was present in several delivery operations (of
the same picking type), the first call to 'scan_deliver' was working, but with
the side-effect to set the `qty_done` for the lines of ALL delivery
operations related to this product.

This fix will process only the lines from one delivery operation at a
time, leaving the other lines untouched so they can be processed by
subsequent calls to `scan_deliver`, without raising the
`"No product found among current transfers."` error anymore (product was
not found because of the `qty_done` set during the first scan).

TDu committed:
shopfloor: checkout no split move line

Disable the splitting of move lines when the quantity is not totally
done.
The diff should be properly taken into account during the put in pack.

OCA-git-bot committed:
shopfloor 14.0.2.9.1

OCA-git-bot committed:
shopfloor 14.0.2.9.2

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (237 of 237 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

TDu committed:
shopfloor: checkout select orther line if not in pack

On `select_line` when a product is scanned and it is not in a package,
the other line not in a package are also selected, but their quantity
done is not changed.

sebalix committed:
shopfloor, delivery: rework 'Process as pre-packaged' option

As the tooltip of the option was describing, its main goal is to
validate the delivery order (and create a backorder if needed) as soon
as a goods is scanned. This could be a package, a product, a packaging
or a lot.

Also, in all cases, when we scan a lot we want to process 1 qty (like
for product scanning), not all the line.

TDu committed:
shopfloor: add prefill qty option on menus

Co-authored-by: Matthieu Méquignon <matthieu.mequignon@camptocamp.com>
Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>
Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

TDu committed:
shopfloor: fix to_float valiator

TDu committed:
shopfloor: checkout add no prefill qty

Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>
Co-authored-by: Matthieu Méquignon <matthieu.mequignon@camptocamp.com>
Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

TDu committed:
shopfloor: cluster picking add no prefill qty

Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>
Co-authored-by: Matthieu Méquignon <matthieu.mequignon@camptocamp.com>
Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

TDu committed:
shopfloor: fix location_content_transfer

TDu committed:
shopfloor: zone picking add no prefill qty

Co-authored-by: Juan Miguel Sánchez Arce <juan.sanchez@camptocamp.com>
Co-authored-by: Matthieu Méquignon <matthieu.mequignon@camptocamp.com>
Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

Commit:
[UPD] Update shopfloor.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.10.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

sebalix committed:
sf, loc. content transfer: refactor unreserve process

sebalix committed:
sf: add a 'stock.unreserve' action

TDu committed:
shopfloor: fix cluster picking noprefill not set

When scanning a location the quantity should be set when the no prefill
quantity option is not enabled.

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (240 of 240 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

TDu committed:
shopfloor: add get work option

TDu committed:
shopfloor: add get work option on location content transfer

TDu committed:
shopfloor: fix zone picking qty done passed

Following the changes on the no prefill quantity option.
The quantity done is passed on the move line record to the frontend, but
not written on the record.
A few places were not updated properly for this.

simahawk committed:
shopfloor: checkout improve lines hooks

* allow to turn off group by location
* ease override for lines manipulation

JuMiSanAr committed:
shopfloor: add new messages

JuMiSanAr committed:
shopfloor: allow scan picking by origin

JuMiSanAr committed:
shopfloor: add move_line_count to package data

JuMiSanAr committed:
shopfloor: expose expiration_date in lot

JuMiSanAr committed:
shopfloor: add stock_picking_progress dependency

shopfloor: fix progress data in shopfloor rests

TDu committed:
shopfloor: add location content transfer diagram

OCA-git-bot committed:
shopfloor 14.0.2.10.1

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.11.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.12.0

TDu committed:
shopfloor: fix tests on rebase

OCA-git-bot committed:
shopfloor 14.0.2.13.0

TDu committed:
shopfloor: fix zone picking scenario

Doing a scan of a location to restrict the search for the next scan on a
product, was not working.

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (250 of 250 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

mmequignon committed:
shopfloor: Add no_operation_found message

mmequignon committed:
shopfloor: action_unreserve - Allow to filter by lot

JuMiSanAr committed:
shopfloor: allow kw args in _create_picking test method

mmequignon committed:
shopfloor: wrong_record message - handle packaging

sebalix committed:
shopfloor, checkout: add a hook for lines packaged

TDu committed:
shopfloor: checkout no-prefill increment only one line

When the no prefill quantity option is set on scanning a barcode only
one line matching the barcode scanned should be incremented.

TDu committed:
shopfloor: imp location content transfer

This part is removed because it is done by `mark_move_line_as_picked`
further down

OCA-git-bot committed:
shopfloor 14.0.2.13.1

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.14.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.15.0

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (255 of 255 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

JuMiSanAr committed:
shopfloor: add shopfloor_user_id to detailed operations tree

JuMiSanAr committed:
shopfloor: update move schema

JuMiSanAr committed:
shopfloor: add move_already_done message

JuMiSanAr committed:
shopfloor: add move data

JuMiSanAr committed:
shopfloor: add new messages for checkout

JuMiSanAr committed:
shopfloor: checkout prevent more than planned

We allow the user to scan units beyond the qty todo,
but we block the option of assigning packages
until the qty is reduced to an allowed value

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.16.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

mt-software-de committed:
shopfloor_reception: allow scan lots

jbaudoux committed:
Added translation using Weblate (German)

TDu committed:
shopfloor: do not propagate user_id to backorder

In the shopfloor workflow the user_id should not be propagated to any
backorder created.

simahawk committed:
shopfloor: cluster_picking full bin action opt-out

Add menu flag to disable the full bin action.
This way the operator is forced to finish the picking
using additional bins when full.

TDu committed:
shopfloor: zp use pack location as sublocation

When scanning a package use the location of that package to filter on
sub location.
So if the package has multiple products, the user will not be required
to scan a location after scanning the product.

TDu committed:
shopfloor: fix message wrong package scanned

When the source package scanned is not valid not need to check its
content for multiple products or lots.

TDu committed:
shopfloor: zp do not propose line from other users

Removing move lines assigned to another user in the list of available
work. Tthe risk of users working on the same line is reduced.

TDu committed:
shopfloor: zp block concurent work on same line

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (258 of 258 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

OCA-git-bot committed:
shopfloor 14.0.2.17.0

TDu committed:
shopfloor: delivery fix sublocation domain

OCA-git-bot committed:
shopfloor 14.0.2.17.1

TDu committed:
shopfloor: imp unmark_move_line_as_picked function

A move line may not be linked to a move, use picking_id from the line.

Extract other move line from picking if they are being worked on by an
other user

OCA-git-bot committed:
shopfloor 14.0.2.17.2

OCA-git-bot committed:
shopfloor 14.0.2.17.3

TDu committed:
shopfloor: fix zp set destination qty zero

Do not allow a user to pick a quantity of zero or less.
When the `no_prefill_qty` is set. A quantity of zero is the default
value and it will still work.

[UPD] Update shopfloor.pot

TDu committed:
shopfloor: fix flappy test

simahawk committed:
shopfloor: checkout allow pack all lines at once

On the select_line screen,
you can now skip the selection and scan a delivery package.

When this happens, the system asks for confirmation.

If you scan again the delivery package,
all lines will be selected and packed at the same time in the same package.

simahawk committed:
shopfloor: add generic search.find

A more flexible way to search for records.

simahawk committed:
shopfloor: checkout use search.find

simahawk committed:
shopfloor: bump 14.0.2.18.0

OCA-git-bot committed:
shopfloor 14.0.2.17.4

mt-software-de committed:
[IMP] shopfloor: refactor cancel_work to offer more hooks for override.

Commit:
[UPD] Update shopfloor.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.19.0

JuMiSanAr committed:
shopfloor_reception: improve scanning of lots

[UPD] Update shopfloor.pot

JuMiSanAr committed:
shopfloor: update messages

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (260 of 260 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

TDu committed:
shopfloor: Add first scan location option on zp

Add a new scenario option `First scan location or pack`.
And implement it on the zone picking scenario.

simahawk committed:
shopfloor: use search.find for scan handlers

TDu committed:
shopfloor: fix location content transfer lines selected

A move line can be in ready state, but the related picking not.
This happens when the shipping policy on the pciking is set to
`one` (When all products are ready).

After this change, move lines from a picking not ready, will not be
offered to work on by the location content transfer scenario.

mmequignon committed:
shopfloor: add allow_rma option

mmequignon committed:
shopfloor: _check_picking - allow multiple states

As shopfloor will handle RMA, we might have to check draft pickings from
time to time. This commit allows to extend the default "assigned" state
in such case.

mmequignon committed:
shopfloor: add return_line_invalid_qty message

mmequignon committed:
shopfloor_reception: Enable allow_rma

mmequignon committed:
shopfloor: search - allow to retrieve done deliveries

OCA-git-bot committed:
shopfloor 14.0.2.20.0

TDu committed:
shopfloor: fix zone picking package as substitue on lines

Use the content of the package (products) and not the product in the
packaging description to search for possible move lines.

TDu committed:
shopfloor: fix zone picking package as substitue on lines

Use the content of the package (products) and not the product in the
packaging description to search for possible move lines.

OCA-git-bot committed:
shopfloor 14.0.2.20.1

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.21.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.22.0

sebalix committed:
sf, checkout: use 'validate_moves' helper in 'done' endpoint

jbaudoux committed:
shopfloor: fix split move lines

The move line must be linked to the new split move before calling
_action_confirm.

In case of a reception picking where the supplier location bypass the
reservation, action_assign is called by _action_config here
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1177
which causes the creation of an unwanted new move line here
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1324
This is why we must relink the existing move line to the new move before
calling _action_confirm.

The new move line was also causing the original picking to change state
to waiting.

OCA-git-bot committed:
shopfloor 14.0.2.23.0

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.24.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

TDu committed:
fixup! shopfloor: fix zone picking package as substitue on lines

mmequignon committed:
shopfloor: action - Add create return helpers

mmequignon committed:
shopfloor: Add move_already_returned message

OCA-git-bot committed:
shopfloor 14.0.2.24.1

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (265 of 265 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

mt-software-de committed:
[IMP] shoploor: use lock actions

mt-software-de committed:
[FIX] shopfloor: fix location content transfer find_work

Before:
Also pickings were found which are already assigned to a user
This leaded to two problems:
- if the qty done was already set to the max qty
  by an other user you always got the error no work
- or you got the same location and the same move lines as the user before

Now:
The find_work method gets locked, before finding a location.
Also it searches now for all assigned pickings, which are not marked as printed
and returns the location of the first open move_line

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
shopfloor 14.0.2.25.0

OCA-git-bot committed:
shopfloor 14.0.2.26.0

simahawk committed:
shopfloor: cluster picking fix action done

Avoid calling '_action_done' on non assigned pickings.

OCA-git-bot committed:
shopfloor 14.0.2.26.1

simahawk committed:
shopfloor: scan_anything allow search by product code

Searching only via barcode is limited.
You might have not have a barcode (eg: only manuf code on a label)
and the search should still work.

simahawk committed:
shopfloor: search.find don't break w/ unsupported types

OCA-git-bot committed:
shopfloor 14.0.2.26.2

simahawk committed:
shopfloor: checkout add show_oneline_package_content option

On the menu item you can decide to enable the option
'show_oneline_package_content'.
Once enabled, on the select line screen,
if the package contains only one line
the lot and the product name will be shown.

The use case it tries to solve:
you have multiple packs in front of you
and you want to be able to scan any of the packs
using the lot or the product.

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.27.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

mt-software-de committed:
[IMP] shopfloor: find_work
Before:
Only pickings were found which are printed = False

Now:
find_work will find pickings which are not assigned or assigned to the current user
also it sorts the pickings by user_id, create_date
this ensures that a already assign picking will be selected a first

OCA-git-bot committed:
shopfloor 14.0.2.28.0

OCA-git-bot committed:
shopfloor 14.0.2.28.1

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (268 of 268 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

OCA-git-bot committed:
shopfloor 14.0.2.29.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

[UPD] Update shopfloor.pot

[UPD] README.rst

OCA-git-bot committed:
shopfloor 14.0.2.30.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.30.1

JuMiSanAr committed:
add allow_alternative_destination menu option

OCA-git-bot committed:
shopfloor 14.0.2.30.2

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (271 of 271 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (272 of 272 strings)

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/es_AR/

TDu committed:
shopfloor: Fix product/lot not found message

Rephrasing the error message when a product or a lot has been found for a
barcode. But there is no picking that matches it in the current context.

[UPD] Update shopfloor.pot

JuMiSanAr committed:
shopfloor: display location operations progress in delivery scenario

JuMiSanAr committed:
shopfloor: fix cluster_picking scan_destination not returning qty done

JuMiSanAr committed:
shopfloor: fix cluster_picking change_lot not returning qty done

OCA-git-bot committed:
shopfloor 14.0.2.30.3

TDu committed:
shopfloor: imp zp scan location or pack first

With the option scan_location_or_pack_first on.
When scanning a location, only the lines without a source package
are offered to be processed.
If none are found, ask for scanning a package.

[UPD] Update shopfloor.pot

TDu committed:
shopfloor: cp add option first scan location or pack

JuMiSanAr committed:
shopfloor: make auto_post menu option available to all scenarios

JuMiSanAr committed:
shopfloor: checkout auto post lines

[UPD] Update shopfloor.pot

JuMiSanAr committed:
shopfloor cluster_picking: improve change_pack_lot endpoint

OCA-git-bot committed:
shopfloor 14.0.2.30.4

JuMiSanAr committed:
sf_single_product_transfer: allow to scan packages on start

JuMiSanAr committed:
sf_single_product_transfer: scan_product with package / location context

JuMiSanAr committed:
sf_single_pack_transfer: display all products in pack

[UPD] Update shopfloor.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.30.5

mt-software-de committed:
[IMP] shopfloor: location content transfer add hook

Make _find_location_move_lines from scan location hookable

OCA-git-bot committed:
shopfloor 14.0.2.31.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

JuMiSanAr committed:
shopfloor: update README

[UPD] README.rst

TDu committed:
shopfloor: cp scan_line remove message

Removed because we always want to jump to the next screen when the right
package is scanned. The scanning of the product or lot will be done
on the next screen.
Also the use of `pick_next_line` when there is an error message does not
make sense.

OCA-git-bot committed:
shopfloor 14.0.2.31.1

[UPD] Update shopfloor.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.32.0

[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.33.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.34.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.35.0

TDu committed:
shopfloor: change zone picking option restriction

Decouple the scenario options

* Unload package at destination
* Collect multiple moves on a same destination package

So they can be selected independently.

Commit:
[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.36.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

JuMiSanAr committed:
shopfloor: rename is_shopfloor_return to is_shopfloor_created

[UPD] Update shopfloor.pot

JuMiSanAr committed:
shopfloor: fix checkout message bug

[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.37.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.38.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.2.39.0

OCA-git-bot committed:
shopfloor 14.0.2.40.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

[UPD] Update shopfloor.pot

OCA-git-bot committed:
shopfloor 14.0.2.41.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

OCA-git-bot committed:
shopfloor 14.0.3.0.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-14.0/wms-14.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-14-0/wms-14-0-shopfloor/

sbejaoui committed:
[MIG] - shopfloor migration 16.0: pre-commit

sbejaoui committed:
[16.0][MIG] - shopfloor

sbejaoui committed:
[MIG] - shopfloor: stock

sbejaoui committed:
[FIX] - fix unit test

lmignon committed:
[MIG] shopfloor: Fix tests

sbejaoui committed:
[FIX] - fix picking type form view

jbaudoux commented:
@sbejaoui Why you dropped the migration scripts ?

sbejaoui commented:
> @sbejaoui Why you dropped the migration scripts ?

Based on the migration [procedure](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0), migration scripts should be about the new version, and the scripts for the old versions should be removed (as it's supposed to have been executed already).

jbaudoux commented:
> > @sbejaoui Why you dropped the migration scripts ?
> 
> Based on the migration [procedure](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0), migration scripts should be about the new version, and the scripts for the old versions should be removed (as it's supposed to have been executed already).

Ah thanks for the info :)

sbejaoui committed:
[REF] ref unload_end for better inheritance

lmignon commented on shopfloor/services/cluster_picking.py:
```diff
@@ -1122,17 +1122,19 @@ def set_destination_all(self, picking_batch_id, barcode, confirmation=False):
-            if all(line.shopfloor_unloaded for line in picking_lines):
-                picking._action_done()
+        for picking in lines.batch_id.picking_ids:
+            picking_lines = lines.filtered(lambda l, p=picking: l.picking_id == p)
+            self._unload_set_picking_to_done(picking, picking_lines)
+
+    def _unload_set_picking_to_done(self, picking, picking_lines):
+        if picking.state == "done":
+            return
+        # We set the picking to done only when the last line is
+        # unloaded to avoid backorders.
+        if all(line.shopfloor_unloaded for line in picking.move_line_ids):
+            picking._action_done()
         if self.work.menu.unload_package_at_destination:
-            lines.result_package_id = False
+            picking_lines.result_package_id = False
```
@sbejaoui It's difficult to know what's the semantic behind this new method '_unload_set_picking_to_done'. In your change, you still iterate multi times n the lines.....



```python
from collections import namedtuple

    def _unload_write_destination_on_lines(self, lines, location):
        lines.write({"shopfloor_unloaded": True, "location_dest_id": location.id})
        lines.package_level_id.location_dest_id = location
        LineIdsByPickingValue =  namedtuple("LINESBYP", ["ids", "all_unloaded_status"])
        bypicking = {} 
        for line in lines:
            if line.picking_id.state == "done":
                continue
            value = lines_by_picking_value[line.picking_id]
            if value:
                value = LineIdsByPickingValue(ids = value.ids + line.ids, all_unloaded_status = value.all_unloaded_status and line.shopfloor_unloaded)
            else:
                value = LineIdsByPickingValue(ids=line.ids, all_unloaded_status=line.shopfloor_unloaded)
            lines_by_picking_value[line.picking_id] = value

        for picking, line_values in lines_by_picking_value.items():
            if line_values.all_unloaded_status:
                picking._action_done()
           if self.work.menu.unload_package_at_destination:
               self.env["stock.move.line"].browse(line_values.ids).write({"result_package_id" : False})
     
```

```

sbejaoui replied:
This refactoring has the purpose of separating the picking validation from the line location update. The idea behind it is to enable the addition of custom behavior during picking validation. Your proposition won't help, as we are still handling all picking in the same method

sbejaoui commented on shopfloor/services/cluster_picking.py:
```diff
@@ -1122,17 +1122,19 @@ def set_destination_all(self, picking_batch_id, barcode, confirmation=False):
-            picking_lines = picking.mapped("move_line_ids")
-            if all(line.shopfloor_unloaded for line in picking_lines):
-                picking._action_done()
+        for picking in lines.batch_id.picking_ids:
```
@lmignon ,I replaced the loop on the lines with a loop on the pickings. I don't understand how it iterates multiple times on the lines

jbaudoux committed:
[FIX] shopfloor: inventory. Set qty & date

jbaudoux committed:
[REF] shopfloor: change lot

Change lot process extracted to the module stock_move_line_change_lot

jbaudoux committed:
[FIX] shopfloor: change lot & inventory

lmignon committed:
[IMP] shopfloor: location_content_transfer: Let Odoo manage the backorder strategy

Prior to this change, when a line was partially picking the  created a new move for the remaining qty. We now let the validation mechanism of odoo create the remaining move since it will take into account the backorder strategy defined on the picking type

jbaudoux commented on shopfloor/services/location_content_transfer.py:
```diff
@@ -764,9 +764,10 @@ def set_destination_line(
+        # If a move line reference a move linked to others move lines (different
+        # locations), we have to split the move to process only the lines related to
+        # the current location.
         move_line._extract_in_split_order({"user_id": self.env.uid})
```
How is that possible as when you scan the location at the start, you isolate the move lines from that location in a split order? So at this stage inside the picking we should only have move lines from that location.

In set_destination_all and set_destination_package, I think such is not done. We should have the same behavior in all 3 cases.

I would have dropped that line also unless you can reproduce the use case.

cc @TDu @sebalix

lmignon replied:
It's required by a test (https://github.com/OCA/wms/pull/679/files#diff-d468159958b3bfcb32afaa09c6a77a69e1811aa92d2abe6d4022f1be66a9ad19R637) but reading the code, It seems that the test is not right since the move_line to process is not selected by using the `scan_location` method and therefore the initial split is not done... I'll fix the test and remove the line.

lmignon replied:
@jbaudoux It's fixed into the last commit https://github.com/OCA/wms/pull/679/commits/dbab97dca13c8402b45275840363f8b54ec7ce06

lmignon committed:
[FIX] shopfloor: location_contant_transfer next operation

When a line is partially processed in the location_content_transfer scenario, the next operation must be the one for the remaining qty. To do it, we need to know where the backorder where the remaining operation has been put to force the selection of the next line to process from the backorder picking

sbejaoui committed:
[FIX] shopfloor: don't edit done move line

rousseldenis commented:
/ocabot migration shopfloor

rousseldenis committed:
[IMP] shopfloor: Restore odoo-test-helper for tests / updated pre-commit

lmignon committed:
[FIX] pylint

lmignon committed:
[FIX] shopfloor: Uses inventory to change package lot

lmignon committed:
[FIX] shopfloor: Properly manage packages into checkout

lmignon committed:
[FIX] shopfloor: cluster_picking:  is_zero tests

When an inventory is created and must be validated, the 'inventory_quantity_set' field is set to True on the quant

lmignon committed:
[FIX] shopfloor: Fix checkout tests

lmignon committed:
[FIX] shopfloor: Fix delivery tests

lmignon committed:
[FIX] shopfloor: Fix location_content_transfer tests

lmignon committed:
[FIX] shopfloor: Fix zone_picking tests

lmignon commented:
@rousseldenis @jbaudoux This one is finally :green_circle: Can we proceed to the merge so I can move forward with others PRs and thereafter forward port  the last changes from 14.0

lmignon commented:
An important element for the future. Shopfloor should be split into several smaller modules (1 per scenario). The current approach is not healthy for maintenance.

lmignon committed:
shopfloor: acsone as contributor

rousseldenis commented on shopfloor/actions/inventory.py:
```diff
@@ -51,8 +51,10 @@ def _get_existing_quant(self, location, product, package=None, lot=None, limit=1
 
-    def _create_draft_inventory(self, location, product, lot=None):
-        quants = self._get_existing_quant(location, product, lot=lot, limit=None)
+    def _create_draft_inventory(self, location, product, package=None, lot=None):
```
This should be completed with owner_id field too.

lmignon replied:
maybe after the forward port from  v14

lmignon commented:
ping @simahawk Can we merge this.... It works in production for months (cluster + location_content) and we want to finalize all the pending PRs on shopfloor  for v16 and forward port the last changes from v14 to have both branches aligned.

rousseldenis commented:
Let's merge this one. It will allow to do enhancements in separate PR's.

/ocabot merge nobump

lmignon committed:
[FIX] shopfloor: zone_picking: Predictable sort

Makes the sorting of stock move line more predictable if move lines share the same attributes

rousseldenis commented:
/ocabot merge nobump

jbaudoux commented on shopfloor/services/location_content_transfer.py:
```diff
@@ -450,7 +450,7 @@ def _find_transfer_move_lines(self, location):
     def _write_destination_on_lines(self, lines, location):
         lines.location_dest_id = location
-        lines.package_level_id.location_dest_id = location
+        lines.package_level_id.picking_id.location_dest_id = location
```
I think this change is not correct and should be reverted

cc @rousseldenis @lmignon @mmequignon

jbaudoux commented on shopfloor/services/location_content_transfer.py:
```diff
@@ -0,0 +1,1194 @@
+    # hook used in module shopfloor_checkout_sync
+    def _write_destination_on_lines(self, lines, location):
+        lines.location_dest_id = location
+        lines.package_level_id.picking_id.location_dest_id = location
```
This line needs fixing. Each package taken from the source location can have their own destination
